smart-contracts/test/test_bridgeBankLock.js

```diff
@@ -433,4 +522,142 @@ describe("Test Bridge Bank", function () {
       ).to.be.revertedWith("Pausable: paused");
     });
   });
+
+  describe("Whitelist", function () {
+    it("should allow the operator to add a token to the whitelist", async function () {
+      // Add the tokens into white list
+      await expect(state.bridgeBank.connect(operator)
+        .updateEthWhiteList(state.token1.address, true))
+        .to.emit(state.bridgeBank, 'LogWhiteListUpdate')
+        .withArgs(state.token1.address, true);
+
+      expect(await state.bridgeBank.getTokenInEthWhiteList(state.token1.address)).to.be.equal(true);
+    });
+
+    it("should not allow user to add a token to the whitelist", async function () {
+      // Add the tokens into white list
+      await expect(state.bridgeBank.connect(userOne)
+        .updateEthWhiteList(state.token1.address, true))
+        .to.be.revertedWith("!operator");
+
+      expect(await state.bridgeBank.getTokenInEthWhiteList(state.token1.address)).to.be.equal(false);
+    });
+
+    it("should allow the operator to remove a token from the whitelist", async function () {
+      // Add the tokens into white list
+      await state.bridgeBank.connect(operator)
+        .updateEthWhiteList(state.token1.address, true)
+        .should.be.fulfilled;
+
+      // Remove the token from whitelist
+      await expect(state.bridgeBank.connect(operator)
+        .updateEthWhiteList(state.token1.address, false))
+        .to.emit(state.bridgeBank, 'LogWhiteListUpdate')
+        .withArgs(state.token1.address, false);
+
+      expect(await state.bridgeBank.getTokenInEthWhiteList(state.token1.address)).to.be.equal(false);
+    });
+
+    it("should not allow user to remove a token from the whitelist", async function () {
+      // Add the tokens into white list
+      await state.bridgeBank.connect(operator)
+        .updateEthWhiteList(state.token1.address, true)
+        .should.be.fulfilled;
+
+      await expect(state.bridgeBank.connect(userOne)
+        .updateEthWhiteList(state.token1.address, false))
+        .to.be.revertedWith("!operator");
+
+      expect(await state.bridgeBank.getTokenInEthWhiteList(state.token1.address)).to.be.equal(true);
+    });
+
+    it("should not allow user to lock ERC20 tokens that are not in whitelist", async function () {
+      await state.token1.connect(userOne).approve(
+        state.bridgeBank.address,
+        state.amount
+      );
+
+      // Attempt to lock tokens
+      await expect(state.bridgeBank.connect(userOne).lock(
+        state.sender,
+        state.token1.address,
+        state.amount
+      )).to.be.revertedWith("Only token in whitelist can be transferred to cosmos");
```

Comment:
We need to change what we expect this to be reverted with as we update the error message to specify which whitelist it should have been a part of.